internal/nitrite.go

```diff
@@ -196,83 +69,27 @@ func Verify(
 		return nil, fmt.Errorf("attestation was generated in debug mode")
 	}
 
-	coseSig := coseSignature{
-		Context:     "Signature1",
-		Protected:   cose.Protected,
-		ExternalAAD: []byte{},
-		Payload:     cose.Payload,
-	}
-
-	sigStruct, err := cbor.Marshal(&coseSig)
+	certificates, err := doc.CheckCertificates(certProvider, verificationTime)
 	if err != nil {
-		return nil, ErrMarshallingCoseSignature
-	}
+		return nil, fmt.Errorf("checking document certificates: %w", err)
 
-	signatureOk := checkECDSASignature(
-		certificates[0].PublicKey.(*ecdsa.PublicKey),
-		sigStruct,
-		cose.Signature,
-	)
+	}
+	if len(certificates) < 1 {
+		return nil, fmt.Errorf("certificates chain is empty")
+	}
```

Comment:
Do I need this check or can I assume that if `document.CheckCertificates` returned without error then this is populated? I thought that was a fair assumption to make in Go

Reply:
In some cases.  In general, docs provide guidance and how you process the data can also help with some ideas.  So for example, if you don't know the length and are iterating to see if an item exists, it doesn't matter.  If you don't know the length, it is unspecified, the operation can succeed and produce an empty slice, it isn't a bad idea to check, else you can be out of bounds.  

Here, since the code is part of the library, we get to choose.  In the current version, we must have at least one item in the slice, so if you think that is always the case, you can document it and make the assumption in functions that call it.  Alternatively, if you don't want to make that grantee you can leave it unspecified and update to something like:
```
switch {
    case err != nil:
      		return nil, fmt.Errorf("checking document certificates: %w", err)
    case len(certificates) == 0:
                return nil, fmt.Errorf("certificates chain is empty")    
}
````

Personally, I would do the check and not make the assumption because the assumption doesn't simplify the code much while additional assumptions can make maintenance more cumbersome.  (Now this is internal to the package so it may not matter that much.)

Reply:
Thanks for weighing in. I left the check